### PR TITLE
9 home page

### PR DESCRIPTION
**Pull request by rawand1998**

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,15 +9,17 @@
     "@mui/icons-material": "^5.0.4",
     "@mui/lab": "^5.0.0-alpha.53",
     "@mui/material": "^5.0.4",
-    "@mui/system": "^5.0.6",
     "@mui/styles": "^5.0.2",
+    "@mui/system": "^5.0.6",
     "axios": "^0.23.0",
     "eslint-config-airbnb": "^18.2.1",
     "eslint-config-airbnb-base": "^14.2.1",
+    "history": "^5.1.0",
```
Comment (LinaYahya) [inline client/package.json:17]: why you install this package?

File: client/src/App/index.js [hunk 2]
```diff
@@ -3,36 +3,56 @@ import './style.css';
...
+        {/* <Navbar /> */}
+        {/* <ul>
+          <li><Link to="/">Home</Link></li>
+          <li><Link to="/cars">cars</Link></li>
+          <li><Link to="/cars/1">car Details</Link></li>
+          <li><Link to="/booking">booking</Link></li>
+          <li><Link to="/dashboard">dashboard</Link></li>
+          <li><Link to="/login">login</Link></li>
+          <li><Link to="/signup">signup</Link></li>
+        </ul> */}
```
Comment (LinaYahya) [inline client/src/App/index.js:27]: Delete commented lines

File: client/src/Pages/Home/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+  };
+  const getTCarsType = async () => {
+    try {
+      const data = await axios.get('/api/v1/types');
+      setTypes(data.data);
+    } catch (err) {
+      console.log(err);
+    }
+  };
```
Comment (LinaYahya) [inline client/src/Pages/Home/index.js:37]: as we're using brands in types in multiple pages, We should provide them to context and use them in the place we need.
and avoid console.log, If an error happens maybe you need to message user with a clear message.

File: client/src/Pages/Home/index.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  useEffect(() => {
+    getTCarsType();
+  }, []);
```
Comment (LinaYahya) [inline client/src/Pages/Home/index.js:40]: all your effects need a cleanup, try to find a way to clean up Axios REST call.

File: client/src/Pages/Home/index.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+      <Box sx={{
+        flexGrow: 1, marginLeft: '3%', marginRight: '3%', height: '600px', paddingTop: '15px',
+      }}
+      >
+        <Grid
+          container
+          spacing={{ xs: 2, md: 3 }}
+          columns={{ xs: 4, sm: 8, md: 12 }}
+        >
+          {brands && brands.map((brand) => (
```
Comment (LinaYahya) [inline client/src/Pages/Home/index.js:92]: what if we had 20 brands, Will we render all of them?!

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/App/index.js (hunk 2 above)
Comment (rawand1998) [reply client/src/App/index.js:27]: done

File: client/src/Pages/Home/index.js (hunk 4 above)
Comment (rawand1998) [reply client/src/Pages/Home/index.js:40]: done

File: client/src/Pages/Home/index.js (hunk 5 above)
Comment (rawand1998) [reply client/src/Pages/Home/index.js:92]: done

File: client/package.json (hunk 1 above)
Comment (rawand1998) [reply client/package.json:17]: I install it when i try to use history and forget to uninstall it.

File: client/src/Pages/Home/index.js (hunk 3 above)
Comment (rawand1998) [reply client/src/Pages/Home/index.js]: Done

**Review by asmaaamin — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Review by AhemdAboAbadi — APPROVED** [review]

**Pull request merged**